lib/map.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const { promisifyClosure } = require('./utils/promisify');
+
+const map = async (fn, arr = [], config = {}) => {
+  const parallel = Object.prototype.hasOwnProperty.call(config, 'parallel') ?
+    config.parallel : false;
```

Comment:
```suggestion
  const parallel = config.parallel;
```
I see no need to use Object.prototype.hasOwnProperty.call there